src/libraries/Microsoft.PowerFx.Core/Public/Engine.cs

```diff
@@ -173,92 +201,115 @@ public CheckResult Check(string expressionText, RecordType parameterType, Parser
         /// <param name="parameterType">types of additional args to pass.</param>
         /// <param name="options">parser options to use.</param>
         /// <returns></returns>
-        public CheckResult Check(ParseResult parse, RecordType parameterType, ParserOptions options = null)
+        [Obsolete("Use other check overload. Shouldn't need both ParserOptions and ParseResult.")]
+        public CheckResult Check(ParseResult parse, RecordType parameterType, ParserOptions options)
+        {
+            return Check(parse, parameterType);
+        }
+
+        [Obsolete("Use other check overload. Shouldn't need both ParserOptions and ParseResult.")]
+        internal CheckResult Check(ParseResult parse, ParserOptions options)
+        {
+            parse.Options = options;
+            return Check(parse, null, options);
+        }
+
+        public CheckResult Check(ParseResult parse, RecordType parameterType = null)
```

Comment:
>Check [](http://example.com/codeflow?start=27&length=5)

We should test parse is not null (or add a comment it's allowed)

Reply:
SetText will catch it.